pkg/cmd/dedicated/listclusters/list_clusters.go

```diff
@@ -0,0 +1,176 @@
+
+func NewListClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   f.Localizer.MustLocalize("dedicated.listClusters.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("dedicated.listClusters.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("dedicated.listClusters.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runListClusters(opts, f)
+		},
+	}
+
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	flags.IntVar(&opts.page, "page", int(cmdutil.ConvertPageValueToInt32(build.DefaultPageNumber)), f.Localizer.MustLocalize("dedicated.listClusters.flag.page"))
+	flags.IntVar(&opts.limit, "size", int(cmdutil.ConvertSizeValueToInt32(build.DefaultPageSize)), f.Localizer.MustLocalize("dedicated.listClusters.flag.limit"))
+
+	return cmd
+}
+
+func runListClusters(opts *options, f *factory.Factory) error {
+	err := listEnterpriseClusters(opts, f)
+	if err != nil {
+		return err
+	}
+	err = getPaginatedClusterList(opts)
+	if err != nil {
+		return err
+	}
+	err = mapRegisteredClusters(opts)
+	if err != nil {
+		return err
+	}
+	err = displayRegisteredClusters(opts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func listEnterpriseClusters(opts *options, f *factory.Factory) error {
+	conn, err := f.Connection()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	api := conn.API()
+	cl := api.KafkaMgmtEnterprise().GetEnterpriseOsdClusters(ctx)
+	clist, response, err := cl.Execute()
+	if err != nil {
+		return err
+	}
+	opts.kfmClusterList = clist
+	f.Logger.Info(response)
+	if len(opts.kfmClusterList.Items) == 0 {
+		return f.Localizer.MustLocalizeError("dedicated.listClusters.log.info.noClusters")
+	}
+	return nil
+}
+
+func getPaginatedClusterList(opts *options) error {
+	cl, err := clustermgmt.GetClusterList(opts.f, opts.accessToken, opts.clusterManagementApiUrl, opts.page, opts.limit)
+	if err != nil {
+		opts.f.Localizer.MustLocalizeError("dedicated.listClusters.cmd.errorGettingClusterList")
+		return err
+	}
+	opts.clustermgmtClusterList = validateClustermgmtClusters(cl, opts.clustermgmtClusterList)
+	return nil
+}
+
+func validateClustermgmtClusters(clusters *clustersmgmtv1.ClusterList, cls []clustersmgmtv1.Cluster) []clustersmgmtv1.Cluster {
+	for _, cluster := range clusters.Slice() {
+		if cluster.State() == clusterReadyState && cluster.MultiAZ() == true {
+			cls = append(cls, *cluster)
+		}
+	}
+	return cls
+}
+
+func mapRegisteredClusters(opts *options) error {
+	for _, cluster := range opts.clustermgmtClusterList {
+		// if cluster is registered, add it to the list of registered clusters
+		if isRegistered, kfmcluster := isClusterRegistered(&cluster, &opts.kfmClusterList); isRegistered {
+			opts.registeredClusters = append(opts.registeredClusters, clusterRow{
+				ID:            cluster.ID(),
+				Name:          cluster.Name(),
```

Comment:
```suggestion
				Name:          cluster.DisplayName(),
```

The `.name` field is static and creation time only. The display name can be changed by the user and we should be using this one instead. Can we verify other location as well and update accordingly?

Reply:
e.g https://github.com/redhat-developer/app-services-cli/blob/bf1a206b76c6be17aa95ed2e793f6235148f83ef/pkg/cmd/dedicated/register/registercluster.go#L144 but in a separate PR to this one

Reply:
DisplayName is not in the OCM SDK as discussed 👍